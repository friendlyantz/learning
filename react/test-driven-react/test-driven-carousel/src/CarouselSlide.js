import React from "react";
import propTypes from "prop-types";

const CarouselSlide = ({ imgUrl, description, attribution, ...rest }) => (
  <figure {...rest}>
    <img src={imgUrl} />
    <figcaption>
      <strong>{description}</strong> {attribution}
    </figcaption>
  </figure>
);

CarouselSlide.propTypes = {
  imgUrl: propTypes.string.isRequired,
  description: propTypes.string.isRequired,
  attribution: propTypes.node
};

export default CarouselSlide;
