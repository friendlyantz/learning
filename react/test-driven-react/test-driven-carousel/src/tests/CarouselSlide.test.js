import React from "react";
import { shallow, mount } from "enzyme";
import CarouselSlide from "../CarouselSlide";
import styled from "styled-components";

describe("CarouseSlide", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CarouselSlide
        imgUrl="https://example.com/default.jpg"
        description="Default test image"
      />
    );
  });

  it("renders correctly", () => {
    wrapper.setProps({
      describe: "Description",
      attribution: "Attribution"
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("passes other props through to the <figcaption>", () => {
    const style = {};
    const onClick = () => {};
    const className = "my-carousel-slide";
    wrapper.setProps({ style, onClick, className });
    expect(wrapper.prop("style")).toBe(style);
    expect(wrapper.prop("onClick")).toBe(onClick);
    expect(wrapper.prop("className")).toBe(className);
  });
});

describe("Img", () => {
  let mounted;
  const imgUrl = "https://example.com/default.jpg";

  beforeEach(() => {
    mounted = mount(
      <CarouselSlide.defaultProps.Img src={imgUrl} imgHeight={500} />
    );
  });

  it("renders correctly", () => {
    expect(mounted.find("img")).toMatchSnapshot();
  });

  it("users imgHeight as the height style property", () => {
    expect(mounted).toHaveStyleRule("height", "500px");
    mounted.setProps({ imgHeight: "calc(100vh - 100px)" });
    expect(mounted).toHaveStyleRule("height", "calc(100vh - 100px)");
  });
});
