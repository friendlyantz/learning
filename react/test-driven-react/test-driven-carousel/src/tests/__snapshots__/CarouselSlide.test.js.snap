// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouseSlide renders correctly 1`] = `
<figure
  describe="Description"
>
  <CarouselSlide__DefaultImg
    imgHeight={500}
    src="https://example.com/default.jpg"
  />
  <figcaption>
    <strong>
      Default test image
    </strong>
     
    Attribution
  </figcaption>
</figure>
`;

exports[`Img renders correctly 1`] = `
.c0 {
  object-fit: cover;
  width: 100%;
  height: 500px;
}

<img
  className="c0"
  src="https://example.com/default.jpg"
/>
`;
